### ISSUE TITLE
Error "Registering Texture" crash.

### ISSUE THREAD

**Issue by VoidLost** [ImpactDevelopment/ImpactIssues#1652]
**1.12.2 | 4.7.3 Impact client crashes after SECOND start with the client.**
First time opening wont cause any problems, second time it wont open.

**To Reproduce**
1. Install Impact 4.7.3 
2. Run game.
3. Quit it.
4. Run again, crashes.

<details> 
  <summary><strong>Crash report</strong></summary>
 

```
---- Minecraft Crash Report ----
// Quite honestly, I wouldn't worry myself about that.

Time: 9/15/19 5:47 AM
Description: Registering texture

java.lang.NullPointerException: Registering texture
	at net.impactclient.2H.a(Unknown Source)
	at cdh.a(ThreadDownloadImageData.java:118)
	at cdr.a(TextureManager.java:90)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah$$Lambda$82/1460670639.apply(Unknown Source)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1118)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah$$Lambda$77/570205651.apply(Unknown Source)
	at java.util.stream.Collectors.lambda$toMap$171(Collectors.java:1321)
	at java.util.stream.Collectors$$Lambda$80/1268894895.accept(Unknown Source)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.2X.0(Unknown Source)
	at net.impactclient.4G.0(Unknown Source)
	at bib.handler$init$zzl000(SourceFile:3136)
	at bib.aq(SourceFile:583)
	at bib.a(SourceFile:404)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at net.impactclient.2H.a(Unknown Source)
	at cdh.a(ThreadDownloadImageData.java:118)

-- Resource location being registered --
Details:
	Resource location: minecraft:downloaded/2378311fabbccae299383fa84b1c8126b4c497ce
	Texture object class: net.impactclient.3M
Stacktrace:
	at cdr.a(TextureManager.java:90)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah$$Lambda$82/1460670639.apply(Unknown Source)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1118)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.0Ah$$Lambda$77/570205651.apply(Unknown Source)
	at java.util.stream.Collectors.lambda$toMap$171(Collectors.java:1321)
	at java.util.stream.Collectors$$Lambda$80/1268894895.accept(Unknown Source)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:512)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:502)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.impactclient.0Ah.0(Unknown Source)
	at net.impactclient.2X.0(Unknown Source)
	at net.impactclient.4G.0(Unknown Source)
	at bib.handler$init$zzl000(SourceFile:3136)
	at bib.aq(SourceFile:583)

-- Initialization --
Details:
Stacktrace:
	at bib.a(SourceFile:404)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 108479664 bytes (103 MB) / 268435456 bytes (256 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	Launched Version: 1.12.2-Impact_4.7.3-OptiFine_HD_U_E3
	LWJGL: 2.9.4
	OpenGL: ASUS R7 370 Series GL version 4.5.13399 Compatibility Profile Context 15.201.1001.0, ATI Technologies Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 8x AMD FX(tm)-8320 Eight-Core Processor 
	OptiFine Version: OptiFine_1.12.2_HD_U_E3
	OptiFine Build: 20181210-121000
	Render Distance Chunks: 12
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.5.13399 Compatibility Profile Context 15.201.1001.0
	OpenGlRenderer: ASUS R7 370 Series
	OpenGlVendor: ATI Technologies Inc.
	CpuCount: 8
```

</details>

<details> 
  <summary><strong>Additional Client Info</strong></summary>
 

- Impact Version: 4.7.3
- Using Optifine? No
- Using LiteLoader? No
- Enabled mods: Does not matter.
- Operating System: Windows 64bit

</details>

**Comment by leijurv** [comment]
Do you have custom main menu on?

**Comment by VoidLost** [comment]
> Do you have custom main menu on?

No

**Comment by VoidLost** [comment]
To fix it, I delete the "assets" folder in Impact. (.minecraft)

**Comment by EverythingIsTakenLOL** [comment]
> 
> 
> **1.12.2 | 4.7.3 Impact client crashes after SECOND start with the client.**
> First time opening wont cause any problems, second time it wont open.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
>     1. Install Impact 4.7.3
> 
>     2. Run game.
> 
>     3. Quit it.
> 
>     4. Run again, crashes.
> 
> 
> **Crash report**
> **Additional Client Info**

I have this same problem, should I just delete the "assets" folder every time? Also what about Forge?

**Comment by rdmir3** [comment]
Duplicate https://github.com/ImpactDevelopment/ImpactIssues/issues/1460 unless you confirm that all of the files in `Impact/assets` are valid PNG files.

**Comment by leijurv** [comment]
Fixed in 4.8
